parkbenchcommon/broadcaster.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2017-2018 Joel Allen Luellwitz and Andrew Klapp
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Provides the broadcasting component of a filesystem-based IPC mechanism."""
+
+__all__ = ['BroadcasterBroadcastException',
+           'BroadcasterInitException',
+           'Broadcaster']
+
+import datetime
+import logging
+import os
+import stat
+import traceback
+import tmpfs
+
+TMPFS_SIZE = '1M'
+
+class BroadcasterBroadcastException(Exception):
+    """This exception is raised when a BeaconBroadcaster object fails to issue a broadcast."""
+
+class BroadcasterInitException(Exception):
+    """This exception is raised when a BeaconBroadcaster object fails to initialize."""
+
+class Broadcaster(object):
+    """Provides the broadcasting component of a filesystem-based IPC mechanism."""
+
+    def __init__(self, program_name, broadcast_name, uid, gid):
+        """Initial configuration of the broadcast directory. This must be done as root. This
+        method creates any necessary spool directories, sets their permissions, and mounts a
+        ramdisk if necessary.
```

Comment:
I know I might have suggested the text on this before but I would like it changed a bit "... This constructor mounts a ramdisk and creates any necessary spool directories with proper permissions."